conan/internal/runner/docker.py

```diff
@@ -64,23 +64,20 @@ def __init__(self, conan_api, command, host_profile, build_profile, args, raw_ar
         import docker
         import docker.api.build
         try:
-            try:
-                self.docker_client = docker.from_env()
-                self.docker_api = docker.APIClient()
-            except:
-                docker_base_urls = [
-                    os.environ.get('CONAN_RUNNER_DOCKER_HOST'),
-                    os.environ.get('DOCKER_HOST'),
-                    'unix:///var/run/docker.sock',
-                    f'unix://{os.path.expanduser("~")}/.rd/docker.sock'
-                ]
-                for base_url in docker_base_urls:
-                    try:
-                        self.docker_client = docker.DockerClient(base_url=base_url, version='auto')
-                        break
-                    except:
-                        continue
-                self.docker_api = self.docker_client.api
+            docker_base_urls = [
+                None, # Default docker configuration, let the python library detect the socket
+                os.environ.get('CONAN_RUNNER_DOCKER_HOST'), # Connect to socket defined in CONAN_RUNNER_DOCKER_HOST
+                os.environ.get('DOCKER_HOST'), # Connect to socket defined in DOCKER_HOST
+                'unix:///var/run/docker.sock', # Default linux socket
+                f'unix://{os.path.expanduser("~")}/.rd/docker.sock' # Rancher linux socket
+            ]
+            for base_url in docker_base_urls:
+                try:
+                    self.docker_client = docker.DockerClient(base_url=base_url, version='auto')
```

Comment:
Some info message, maybe at least ``ConanOutput().verbose()`` of the things that are being tried would make sense.